File: Leetcode_with_py/07reverse.py
class Solution(object):
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        if x==0:
            return 0
        str_x = str(x)
        x = ''
        if str_x[0] == '-':
            x += '-'
        x += str_x[::-1].lstrip("0").rstrip("-")
        x = int(x)
        if -2**31<x<2**31-1:  # 反转后整数溢出返回 0
            return x
        return 0

File: Leetcode_with_Cpp/07reverse.cpp
class Solution {
public:
    const long MAX = (1<<31)-1, MIN = (-1<<31);  // 最大最小值
    int reverse(int x) {
        if(x/10 == 0) return x;  // 绝对值小于10，直接返回
        long y = 0;
        while(x) {
            y *= 10;
            if(y > MAX || y < MIN) 
                return 0;
            y += x % 10;  // 取出x的个位
            x /= 10;      // 去掉x的个位
        }
        return y;
    }
};